Remotion/Validation/UnitTests/Rules/ObjectValidationRuleTest.cs

```diff
@@ -1,21 +1,128 @@
-﻿using NUnit.Framework;
+﻿using System;
+using System.Collections.Generic;
+using Moq;
+using NUnit.Framework;
+using Remotion.Validation.Results;
+using Remotion.Validation.Rules;
+using Remotion.Validation.Validators;
 
 namespace Remotion.Validation.UnitTests.Rules
 {
   [TestFixture]
   public class ObjectValidationRuleTest
   {
     [Test]
-    [Ignore("TODO RM-5906")]
-    public void Validate ()
+    public void Validate_ReturnsValidationFailures ()
     {
+      Func<int, bool> condition = i => true;
+      var context = new ValidationContext(1);
+
+      var failure1 = new ObjectValidationFailure(
+          15,
+          "15 is not a number",
+          "Localized validation message: 15 is not a number.");
+
+      var failure2 = new ObjectValidationFailure(
+          42,
+          "42 is not a number",
+          "Localized validation message: 42 is not a number.");
+
+      var validatorMock1 = new Mock<IObjectValidator>();
+      validatorMock1.Setup(_ => _.Validate(It.IsAny<ObjectValidatorContext>()))
```

Comment:
If we use 
```suggestion
      validatorMock1.Setup(_ => _.Validate(context))
```
the code is just as easy to maintain/robust and more compact. it also ensured that we don't pass null instead of the actual object.

Reply:
Sadly, this doesnt work. The context for the object validators is created inside the rule, we cannot create it beforehand and pass it through.

Reply:
That's true - different objects. My bad.

So, updated request: we should have a test that verifies that ObjectValidationContext is correctly constructured. we can do the following:

ObjectValidatioNContext actualCtx
validatorMock2.Setup(_ => _.Validate(It.IsAny<ObjectValidatorContext>()))
          .WhenCalled(ctx => actualCtx = ctx)
          .Returns(new[] { failure2 })
          .Verifiable();
Assert.that (actualCtx, Is.Not.Null)
Assert.that(actualCtx.Parent, Is.SameAs(context))
Assert.that(actualCtx.InstanceToValidate, Is.SameAs(contxt.InstanceToValidate))

Doing it this way instead of just having a predicate in It.Is<ObjectValidationContext>() decouples the mock from the assertion (always nice when done cheaply) and gives us better diagnostic when there do is an error. And it also hihglihgts the actual test we want to do (here: test that the ObjctValidatioNContext is built correctly)